File: wwwroot/js/models/inventoryItem.js
var InventoryItem = function (data) {
    var self = this;
    if (data) {
        ko.mapping.fromJS(data, {}, self);
    }
    else {
        self.date = ko.observable(new Date());
        self.itemType = ko.observable();
        self.quantity = ko.observable();
    }
}

File: Domain/InventoryItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace MessingSystem.Domain
{
    public class InventoryItem
    {
        public int InventoryItemId { get; set; }
        public DateTime? Date { get; set; }
        public int ItemType { get; set; }
        public decimal Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public bool IsDeleted { get; set; }

    }
}
